### PR TITLE
Upgrade @guardian/tsconfig and friends

### PR DESCRIPTION
**Pull request by jamesgorrie**
Upgrade `@guardian/tsconfig` to v1.0.0.
https://github.com/guardian/csnx/blob/main/libs/%40guardian/tsconfig/CHANGELOG.md#100

We had a few Types we had to update on the way, and also had to update `preact`.

# this branch
Screenshot 2024-06-06 at 14 49 51

# main
Screenshot 2024-06-06 at 15 07 43

# this branch 
Screenshot 2024-06-06 at 14 50 54

# main
Screenshot 2024-06-06 at 15 07 33

### CODE REVIEW

File: support-frontend/assets/helpers/redux/checkout/product/reducer.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import type { PayloadAction } from '@reduxjs/toolkit';
 import { createSlice } from '@reduxjs/toolkit';
-import type { WritableDraft } from 'immer/dist/types/types-external';
+import type { Draft } from 'immer';
```
Comment (jamesgorrie) [inline support-frontend/assets/helpers/redux/checkout/product/reducer.ts:3]: [As per](https://github.com/immerjs/immer/issues/726#issuecomment-752876708)

File: support-frontend/assets/helpers/redux/utils/validation/errors.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import type { PayloadAction } from '@reduxjs/toolkit';
-import type { WritableDraft } from 'immer/dist/internal';
+import type { Draft } from 'immer';
```
Comment (jamesgorrie) [inline support-frontend/assets/helpers/redux/utils/validation/errors.ts:2]: [As per](https://github.com/immerjs/immer/issues/726#issuecomment-752876708)

**Comment by GHaberis** [comment]
@jamesgorrie can we do a page speed performance check on this branch vs `main`, last week we reverted an update to `preact` as it led to a notable slow down in rendering times: https://github.com/guardian/support-frontend/pull/6031

**Comment by jamesgorrie** [comment]
@GHaberis - I've added the results. They seem marginally better in this PR.

File: support-frontend/assets/helpers/rendering/render.ts [hunk 3]
```diff
@@ -46,19 +46,20 @@ const renderError = (e: Error): void => {
...
 
 	if (element) {
 		delete element.dataset.notHydrated;
 
 		try {
 			if (process.env.NODE_ENV === 'development' && !isSafari) {
-				// @ts-expect-error - Not sure why it's not finding typedefs for Preact
 				void import('preact/debug');
-				void import('@axe-core/react').then((axe) => {
+				void Promise.all([
```
Comment (GHaberis) [inline support-frontend/assets/helpers/rendering/render.ts:58]: @jamesgorrie are you able to run the site in Dev mode locally? I'm seeing an issue in the client-side bundle that looks like it's related to the updates in this file.

Screenshot 2024-06-06 at 17 31 56

Comment (jamesgorrie) [reply support-frontend/assets/helpers/rendering/render.ts]: Hmm - yea, I've removed that refactor as it's not actually part of this and didn't reduce the package size either...

**Review by GHaberis — APPROVED** [review]
Looks good

**Pull request merged**